### ISSUE TITLE
Hotswapping with Ant doesn't seem to be working

### ISSUE THREAD

**Issue by ameyp** [skeeto/sample-java-project#1]
I'm using JDK 7 (1.7u21) on 32-bit Linux, along with Ant 1.9.2 and Ivy 2.3.0, and the hotswap example here doesn't work. The hotswap-demo branch of october-chess-engine isn't working either. If you want I can provide console logs for running `ant run` and `ant hotswap` (and `ant run-hotswap` for october-chess-engine) but essentially the ant target succeeds and nothing changes in the running program.

I don't know if you're still working with Java, but I would really appreciate some help, or even a nudge in the right direction.

**Comment by skeeto** [comment]
I'm sorry, it looks like at some point I broke the hotswap demo. This should be fixed by 1f633b4. It works for me with both OpenJDK 1.6.0_27 and 1.7.0_25.

With hotswap when a method is updated it's only used next time the method called. If the original method is already on the stack or otherwise in use, it will continue to run unchanged (i.e. the original version finishes its course). The problem was that the `"Hello, %s"` string was being printed in a loop within the same method. When the `run()` method was replaced by hotswap, the _new_ version of the method was never actually run because it was never invoked. Instead just the old loop continued to execute. I moved the print statement out into its own method, `sayHello()`. When a new version is hotswapped, it will be invoked next time around the loop.

Ad for october-chess-engine, keep in mind that hotswapping has all sorts of caveats. You can't change any class or method signatures, but also it's finicky about constants. For example, in BoardPanel.java, you could try to change the constants `DARK` and `LIGHT` and hotswap, but it probably won't work. I think the JVM inlines these or something. Instead, change the `paintComponent()` method where these constants are being used (lines 217 and 219). After you hotswap you also need to force a redraw, which you can do by clicking anywhere on the chess board in the GUI, or by having the computer play itself while you hotswap.

```
if ((x + y) % 2 == 0) {
    g.setColor(Color.WHITE);
} else {
    g.setColor(Color.BLACK);
}
```

If you hotswap the above and click the board, you should see the board go from the default brown shades to straight black and white.

**Comment by ameyp** [comment]
Thanks. In retrospect it makes sense that someone would have to call the modified function for the changes to be reflected, don't know why that didn't occur to me. And the chess engine also works when I click on the board, I just assumed that the redraw would happen every second or so, based on your blog post. And now I realize that I had forgotten to checkout the `hotswap-demo` branch.

Thanks for your help!
